### PR TITLE
Alerting: don't copy original rule uid when cloning

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

Currently we're unable to clone rules because the cloned one is being assigned the original rule's uid, giving an error when saving the form.

**Why do we need this feature?**

To be able to clone rules.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Reported on Slack.

**Special notes for your reviewer:**

Before

After

### CODE REVIEW

**Review by gillesdemey — APPROVED** [review]
Code LGTM 🚢

File: public/app/features/alerting/unified/CloneRuleEditor.tsx [hunk 1]
```diff
@@ -42,6 +42,9 @@ export function CloneRuleEditor({ sourceRuleId }: { sourceRuleId: RuleIdentifier
       formPrefill.group = '';
     }
 
+    // Since this is a new rule, we don't want to carry over the uid of the original rule
+    formPrefill.uid = '';
```
Comment (konrad147) [inline public/app/features/alerting/unified/CloneRuleEditor.tsx:46]: Can we add a test for that? The thing My concern is that the cloning became broken due to changes in other components, not in this one

**Comment by konrad147** [comment]
Could you provide more details about what was broken? The file containing the fix hasn't been changed in months, so changes in some external components had to lead to the issue

**Comment by soniaAguilarPeiron** [comment]
> Could you provide more details about what was broken? The file containing the fix hasn't been changed in months, so changes in some external components had to lead to the issue

I would say that this is because after adding the notification preview in the alert rule form, we needed to add this alert `uid` as a prop in the form, and as a consecuence we had to have it in the `rulerRuleToFormValues`. As prefill was using these  `rulerRuleToFormValues` to prefill we need to remove this uid value to not use the original one from where all the values are copied.

**Comment by gillesdemey** [comment]
To move forward with this PR I suggest creating a `cloneRuleDefinition` function that 

1. Does a `cloneDeep` of the rule
2. Calls the `changeRuleName` function to create the copied rule name
3. omits the UID
4. omits the group if the original rule has a provenance (provisioned)

We can pass the result of that function to `rulerRuleToFormValues()` without having to do any additional mutations.

We can then write a test for it. Regardless of why the UID is now in the form it's probably best to be explicit about omitting it from the cloned rule.

**Comment by soniaAguilarPeiron** [comment]
> 
I reviewed the code, and I realised we can get rid of this new parameter we added because of the notifications preview, as we have it on the url. So, we can remove it and use the url param instead to avoid more problems like the one we have in here. I tested in my local and by doing this the clone issue disappears. Wdyt @gillesdemey ?

**Comment by gillesdemey** [comment]
I think that's a great approach, but would also like to see what I've outlined above as that

1. prevents future regressions if UID somehow slips back in to the form
2. makes the current clone behaviour testable and tested

That's not mutually exclusive with your proposal, so we can do both :)

**Comment by soniaAguilarPeiron** [comment]
> I think that's a great approach, but would also like to see what I've outlined above as that
> 
> 1. prevents future regressions if UID somehow slips back in to the form
> 2. makes the current clone behaviour testable and tested
> 
> That's not mutually exclusive with your proposal, so we can do both :)

💯

File: public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const AlertRuleForm = ({ existing, prefill }: Props) => {
 
   const routeParams = useParams<{ type: string; id: string }>();
   const ruleType = translateRouteParamToRuleType(routeParams.type);
-  const uid_fromparams = routeParams.id;
+  const uidFromParams = routeParams.id;
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:87]: 💯

**Review by gillesdemey — APPROVED** [review]
Great work! 🚀

**Pull request merged**